### DISCUSSION TITLE
postUpgradeTasks and missing Docker socket

### DISCUSSION THREAD

**Discussion by mtardy** [renovatebot/renovate#22379] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.98.4

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I try to use the `postUpgradeTasks` feature to run a command in order to generate files and it calls Docker, apparently, it fails to find the Docker socket. Is it possible to use Docker command there since renovate is running via Docker?

I'm running self-hosted on GitHub actions, here is an example of PR postUpgradeTasks failure https://github.com/mtardy/tetragon/pull/17#issuecomment-1559679171. My test configuration is there https://github.com/mtardy/tetragon/blob/main/.github/renovate.json5.

### What you tried so far.

I tried installing buildx and it still outputs the same thing, but I don't think that's the main issue there. I can still find a way to run `docker build` instead of buildx but I guess it won't solve the missing socket issue.

I tried without the buildx command using just `docker build` and same result:
```
Command failed: make codegen
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
make: *** [Makefile:252: image-codegen] Error 1
```

Here are more details with debug enabled: https://github.com/mtardy/tetragon/actions/runs/5059753721/jobs/9081738168#step:5:8296

### Relevant debug logs

```
Command failed: make codegen
docker: 'buildx' is not a docker command.
See 'docker --help'
failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial unix /var/run/docker.sock: connect: no such file or directory
make: *** [Makefile:252: image-codegen] Error 1
```

**Comment by viceice** [comment, marked as answer]
you need to mount the docker socket to the renovate container

**Reply by mtardy** [reply]
Hey thanks! How can I do that in the context of using the renovate bot GitHub action? I guess for this use case I have to run renovate without the GitHub action and directly using Docker?

**Reply by mtardy** [reply]
I was going to propose a PR to the GitHub action to add it as an option but you just removed the docker socket mount so maybe it's not in your interest. https://github.com/renovatebot/github-action/pull/700. Please tell me if you think it's a good idea :)

**Reply by viceice** [reply]
you can propose a feature PR to optionally mount the docker socket to the container

**Reply by mtardy** [reply]
Here it is https://github.com/renovatebot/github-action/pull/749 😺
